src/net/rdma.cc

```diff
@@ -752,7 +766,6 @@ std::unique_ptr<RDMAConnection> RDMAStack::connect(const EndPoint& remote) {
 void RDMAStack::processUDMessage(UDMessage* message, EndPoint remote) {
     if (message->op == UDOps::HandshakeRequest) {
         std::unique_ptr<RDMAConnection> conn = std::make_unique<RDMAConnection>(this, remote);
-        handshakeLookup[message->requestId] = conn->weak_from_this();
```

Comment:
Not sure if this is correct. I think this was supposed to enter the entry into the map on incoming HS request. We do the same on out-going. I think we just have to make sure that if one exists, we don't process but instead return the existing one, right?

Reply:
Checked with Justin. This map only holds outgoing handshake info so it should not be updated on incoming handshake.